trio/_core/_multierror.py

```diff
@@ -328,10 +328,7 @@ def catch(
         return MultiErrorCatcher(handler)
 
 
-if TYPE_CHECKING:  # noqa: SIM108
-    _ExceptionGroup = ExceptionGroup[Exception]
-else:
-    _ExceptionGroup = ExceptionGroup
+_ExceptionGroup = ExceptionGroup[Exception] if TYPE_CHECKING else ExceptionGroup
```

Comment:
FYI latest ruff will not do this, so this change is fine even if type checkers don't like it.

Reply:
Latest ruff does still apply interestingly enough, so I added a noqa for it.

Reply:
Fixed in dd7b0ed